oracle/feed_stork.go

```diff
@@ -0,0 +1,260 @@
+	ProviderName string `toml:"provider"`
+	Ticker       string `toml:"ticker"`
+	PullInterval string `toml:"pullInterval"`
+	Message      string `toml:"message"`
+	OracleType   string `toml:"oracleType"`
+}
+
+type storkPriceFeed struct {
+	providerName string
+	interval     time.Duration
+	message      string
+
+	logger  log.Logger
+	svcTags metrics.Tags
+
+	oracleType oracletypes.OracleType
+}
+
+func ParseStorkFeedConfig(body []byte) (*StorkFeedConfig, error) {
+	var config StorkFeedConfig
+	if err := toml.Unmarshal(body, &config); err != nil {
+		err = errors.Wrap(err, "failed to unmarshal TOML config")
+		return nil, err
+	}
+
+	return &config, nil
+}
+
+// NewStorkPriceFeed returns price puller
+func NewStorkPriceFeed(cfg *StorkFeedConfig) (PricePuller, error) {
+	pullInterval := 1 * time.Minute
+	if len(cfg.PullInterval) > 0 {
+		interval, err := time.ParseDuration(cfg.PullInterval)
+		if err != nil {
+			err = errors.Wrapf(err, "failed to parse pull interval: %s (expected format: 60s)", cfg.PullInterval)
+			return nil, err
+		}
+
+		if interval < 1*time.Second {
+			err = errors.Wrapf(err, "failed to parse pull interval: %s (minimum interval = 1s)", cfg.PullInterval)
+			return nil, err
+		}
+
+		pullInterval = interval
+	}
+
+	var oracleType oracletypes.OracleType
+	if cfg.OracleType == "" {
+		oracleType = oracletypes.OracleType_PriceFeed
+	} else {
+		tmpType, exist := oracletypes.OracleType_value[cfg.OracleType]
+		if !exist {
+			return nil, fmt.Errorf("oracle type does not exist: %s", cfg.OracleType)
+		}
+
+		oracleType = oracletypes.OracleType(tmpType)
+	}
+
+	feed := &storkPriceFeed{
+		providerName: cfg.ProviderName,
+		interval:     pullInterval,
+		message:      cfg.Message,
+		oracleType:   oracleType,
+
+		logger: log.WithFields(log.Fields{
+			"svc":      "oracle",
+			"dynamic":  true,
+			"provider": cfg.ProviderName,
+		}),
+
+		svcTags: metrics.Tags{
+			"provider": cfg.ProviderName,
+		},
+	}
+
+	return feed, nil
+}
+
+func (f *storkPriceFeed) Interval() time.Duration {
+	return f.interval
+}
+
+func (f *storkPriceFeed) Symbol() string {
+	return ""
+}
+
+func (f *storkPriceFeed) Provider() FeedProvider {
+	return FeedProviderStork
+}
+
+func (f *storkPriceFeed) ProviderName() string {
+	return f.providerName
+}
+
+func (f *storkPriceFeed) OracleType() oracletypes.OracleType {
+	return oracletypes.OracleType_Stork
+}
+
+// PullAssetPairs pulls asset pair for an asset id
+func (f *storkPriceFeed) PullAssetPairs(conn *websocket.Conn) (assetPairs []*oracletypes.AssetPair, err error) {
+	metrics.ReportFuncCall(f.svcTags)
+	doneFn := metrics.ReportFuncTiming(f.svcTags)
+	defer doneFn()
+
+	err = conn.WriteMessage(websocket.TextMessage, []byte(f.message))
+	if err != nil {
+		log.Infoln("Error writing message:", err)
+		return
+	}
+
+	var msgNeed []byte
+	var messageRead []byte
+	count := 0
+	for count < 2 {
+		_, messageRead, err = conn.ReadMessage()
+		if err != nil {
+			log.Infoln("Error reading message:", err)
```

Comment:
We should use the logger that has more context
```suggestion
			f.logger.Warningln("Error reading message:", err)
```